[ID1000375] implement the block chain proof picture components

Nixiangyu opened the pull request:
--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

Commit:
[ID1000375] implement the block chain proof picture components

--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

Commit:
[ID1000375] implement the block chain proof picture components

--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

Commit:
[ID1000375] implement the block chain proof picture components

--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

Commit:
[ID1000375] implement the block chain proof picture components

--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

hukunge commented on components/image/image.go:
```diff
@@ -0,0 +1,170 @@
+		var qrCodeImg image.Image
+		qrCodeImg, err = c.CreateQrCode("http://www.baidu.com")
+		draw.Draw(c.Bg, qrCodeImg.Bounds().Add(image.Pt(100, 710)), qrCodeImg, image.Point{X: 0, Y: 0}, draw.Over)
+		err = c.DrawText(rest.Color2, "长安识别二维码", freetype.Pt(235, 828), 22)
```
“长安”应该改成“长按”吧？应该放到一个常量里

Nixiangyu replied:
好的

hukunge commented on common/rest/const.go:
```diff
@@ -57,3 +59,45 @@ const (
 )
+
+// define the default HTTP header
+const (
```
这是干啥用的

Nixiangyu replied:
图片域名前缀

hukunge commented on components/image/image.go:
```diff
@@ -0,0 +1,170 @@
+	qrCode, err = qrcode.New(content, qrcode.Highest)
+
+	if err != nil {
+		return nil, errors.New("创建二维码失败")
```
英文注释？

Nixiangyu replied:
我修改一下

Nixiangyu committed:
Merge branch 'master' into master

zhangbing16 committed:
Merge branch 'master' into master

zhangbing16 commented on common/rest/const.go:
```diff
@@ -64,3 +66,45 @@ const (
+
+// define the default HTTP header
+const (
+	ArxanChainHTTP = "https://boxdev.arxanchain.com"
```
这个域名的用途是什么？如有用途建议使用配置文件，方便运营等修改配置。

Nixiangyu replied:
好的

zhangbing16 commented on components/image/image.go:
```diff
@@ -0,0 +1,170 @@
+		var qrCodeImg image.Image
+		qrCodeImg, err = c.CreateQrCode("http://www.baidu.com")
+		draw.Draw(c.Bg, qrCodeImg.Bounds().Add(image.Pt(100, 710)), qrCodeImg, image.Point{X: 0, Y: 0}, draw.Over)
+		err = c.DrawText(rest.Color2, "长安识别二维码", freetype.Pt(235, 828), 22)
```
长按

Nixiangyu replied:
好的

zhangbing16 reviewed (commented):
请查看comment信息。

Commit:
[ID1000375] implement the block chain proof picture components

--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

hukunge commented on components/image/image.go:
```diff
@@ -0,0 +1,170 @@
+		var qrCodeImg image.Image
+		qrCodeImg, err = c.CreateQrCode("http://www.baidu.com")
+		draw.Draw(c.Bg, qrCodeImg.Bounds().Add(image.Pt(100, 710)), qrCodeImg, image.Point{X: 0, Y: 0}, draw.Over)
+		err = c.DrawText(rest.Color2, "长按识别二维码", freetype.Pt(235, 828), 22)
```
"长按识别二维码"，建议改成常量

hukunge committed:
[ID1000373] unit test of supplies (#49)

--story=1000373 --user=hukunge
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000373

Signed-off-by: hukunge <hukunge@arxanchain.com>

hukunge committed:
[ID1000336] unit test of wechat app login (#50)

--story=1000336 --user=hukunge
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000336

Signed-off-by: hukunge <hukunge@arxanchain.com>

hukunge commented on components/image/image.go:
```diff
@@ -0,0 +1,157 @@
+)
+
+var (
+	logger = log.MustGetLogger("components/image")
```
logger = log.MustGetLogger("image")

hukunge commented on components/image/image.go:
```diff
@@ -0,0 +1,157 @@
+}
+
+// CreateQrCode  create a wx qr code
+func (c *Client) CreateWXQrCode(appID string, secret string, scene string) (img image.Image, err error) {
```
build failed:
comment on exported method Client.CreateWXQrCode should be of the form "CreateWXQrCode ...

hukunge commented on components/wx/impl.go:
```diff
@@ -12,24 +12,34 @@ import (
 	wxAddress      = "https://api.weixin.qq.com"
 	checkFingerURL = "%s/cgi-bin/soter/verify_signature?access_token=%s"
 	accessTokenURL = "%s/cgi-bin/token?grant_type=client_credential&appid=%s&secret=%s"
+	wxacodeURL     = "https://api.weixin.qq.com/wxa/getwxacodeunlimit?access_token=%s"
```
wxacodeURL中base url部分使用wxAddress

hukunge commented on components/wx/impl.go:
```diff
@@ -186,3 +196,43 @@ func GetAccessToken(appID string, secret string) (string, error) {
 }
+
+// GetWXACode ...
+func GetWXACode(token string, scene string) (image.Image, error) {
```
GetWQRACode

hukunge commented on components/wx/impl.go:
```diff
@@ -186,3 +196,43 @@ func GetAccessToken(appID string, secret string) (string, error) {
+func GetWXACode(token string, scene string) (image.Image, error) {
+	wxURL := fmt.Sprintf(wxacodeURL, token)
+	logger.Debugf("Get wx code url: %v", wxURL)
+	var req = &GetWXACodeRequest{
```
GetWXQRCodeRequest

hukunge commented on components/wx/wx.go:
```diff
@@ -81,3 +81,12 @@ type PhoneNumber struct {
 }
+
+// GetWXACodeRequest ...
+type GetWXACodeRequest struct {
```
GetWXQRCodeRequest

Commit:
[ID1000375] implement the block chain proof picture components

--story=1000375 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000375

Signed-off-by: Nixiangyu nixiangyu@ofworld.com

Commit:
Merge remote-tracking branch 'upstream/master'
